### PR TITLE
chore(web-components): use predicate function instead of instanceof in Accordion

### PR DESCRIPTION
**Pull request by radium-v**
## Previous Behavior

In systems which bundle components individually, we noticed that by using `instanceof` in the `Accordion` class, we're pulling the `BaseAccordion` class into our bundles.

## New Behavior

This PR introduces type predicate functions, `isAccordionItem` and `isAccordion`, to cheaply determine if a child component is one which is expected. These functions do this by checking if the `tagName` ends with a `baseName` value, so `isAccordionItem(element)` will return true if `element.tagName` ends with "accordion-item".

Some other changes:
* The top-level `index.ts` exports `accordionDefinition` and `accordionItemDefinition`. These should be deprecated and removed in favor of the more consistent PascalCased `AccordionDefinition` and `AccordionItemDefinition`.
* The `Accordion.accordionItems` property was typed as `Element[]`, which required a lot of checking to confirm if the collection actually contains `AccordionItem[]`. This PR fixes the property's type to make it easier to work with.

This PR also fixes some minor API Extractor warnings generated by the `Accordion` and `AccordionItem` modules.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: packages/web-components/src/accordion/accordion.definition.ts [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+
+/**
+ * Checks if an element is an {@link Accordion}.
+ *
+ * @param element - The element to check
+ * @returns - The element is an Accordion
+ *
+ * @public
+ */
+export function isAccordion(element: Element): element is Accordion {
```
Comment (marchbox) [inline packages/web-components/src/accordion/accordion.definition.ts:34]: Should `element` be `HTMLElement` instead?

**Review by marchbox — APPROVED** [review]

File: packages/web-components/src/accordion/index.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+  baseName as AccordionBaseName,
+  definition as AccordionDefinition,
+  isAccordion,
+} from './accordion.definition.js';
 export { Accordion } from './accordion.js';
 export { AccordionExpandMode } from './accordion.options.js';
-export { template as accordionTemplate } from './accordion.template.js';
 export { styles as accordionStyles } from './accordion.styles.js';
-export { definition as accordionDefinition } from './accordion.definition.js';
+export { template as accordionTemplate } from './accordion.template.js';
```
Comment (davatron5000) [inline packages/web-components/src/accordion/index.ts:9]: Capitalize `AccordionStyles` and `AccordionTemplate` - Also in PR #32075

File: packages/web-components/src/accordion/accordion.ts [hunk 3]
```diff
@@ -79,28 +79,19 @@ export class Accordion extends FASTElement {
...
-    if (this.accordionItems.length === 0) {
-      return null;
-    }
-
-    return (
-      this.accordionItems.find(
-        (item: Element | BaseAccordionItem) => item instanceof BaseAccordionItem && item.expanded,
-      ) ?? this.accordionItems[0]
-    );
+  private findExpandedItem(): BaseAccordionItem | null {
```
Comment (davatron5000) [inline packages/web-components/src/accordion/accordion.ts:87]: I think based on the changes this can't return `null` anymore, only `undefined`?

**Review by davatron5000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**